lib/gltf.js

```diff
@@ -154,185 +43,301 @@ function createGltf(data, inputPath, modelName) {
     };
 
     gltf.asset = {
-        "generator": "OBJ2GLTF",
-        "premultipliedAlpha": true,
-        "profile": {
-            "api": "WebGL",
-            "version": "1.0"
+        generator : 'obj2gltf',
+        profile : {
+            api : 'WebGL',
+            version : '1.0'
         },
-        "version": 1
+        version: '1.0'
     };
 
     gltf.scenes[sceneId] = {
-        nodes : [nodeId]
+        nodes : []
     };
 
-    gltf.nodes[nodeId] = {
-        children : [],
-        matrix : [1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1],
-        meshes : [meshId],
-        name : modelName
-    };
+    function getImageId(imagePath) {
+        return path.basename(imagePath, path.extname(imagePath));
+    }
 
-    gltf.samplers[samplerId] = {}; // Use default values
-
-    var bufferSeparate = false;
-    var bufferUri;
-    if (buffer.length > 201326580) {
-        // toString fails for buffers larger than ~192MB. Instead save the buffer to a .bin file.
-        // Source: https://github.com/nodejs/node/issues/4266
-        bufferSeparate = true;
-        bufferUri = modelName + '.bin';
-    } else {
-        bufferUri = 'data:application/octet-stream;base64,' + buffer.toString('base64');
+    function getTextureId(imagePath) {
+        if (!defined(imagePath) || !defined(images[imagePath])) {
+            return undefined;
+        }
+        return 'texture_' + getImageId(imagePath);
     }
 
-    gltf.buffers[bufferId] = {
-        byteLength : bufferByteLength,
-        type : 'arraybuffer',
-        uri : bufferUri
-    };
+    function createMaterial(material, hasNormals) {
+        var ambient = defaultValue(defaultValue(getTextureId(material.ambientColorMap), material.ambientColor), [0, 0, 0, 1]);
```

Comment:
If this code is called only a few times, then it's fine, but if it's called thousands of times then we should not used `defaultValue` because it creates a ton of throw-away arrays (remember `defaultValue` can't short circuit so all args get created).  If these arrays are never modified, then it would be a good idea to extract the defaults to module scope variables in either case.